crates/egui/src/widgets/drag_value.rs

```diff
@@ -371,18 +371,40 @@ impl<'a> Widget for DragValue<'a> {
         let shift = ui.input().modifiers.shift_only();
         let is_slow_speed = shift && ui.memory().is_being_dragged(ui.next_auto_id());
 
+        let kb_edit_id = ui.next_auto_id();
+        // The following call ensures that when a `DragValue` receives focus,
+        // it is immediately rendered in edit mode, rather than being rendered
+        // in button mode for just one frame. This is important for
+        // screen readers.
+        ui.memory().interested_in_focus(kb_edit_id);
+        let is_kb_editing = ui.memory().has_focus(kb_edit_id);
+
         let old_value = get(&mut get_set_value);
-        let value = clamp_to_range(old_value, clamp_range.clone());
-        if old_value != value {
-            set(&mut get_set_value, value);
-        }
+        let mut value = old_value;
         let aim_rad = ui.input().aim_radius() as f64;
 
         let auto_decimals = (aim_rad / speed.abs()).log10().ceil().clamp(0.0, 15.0) as usize;
         let auto_decimals = auto_decimals + is_slow_speed as usize;
-
         let max_decimals = max_decimals.unwrap_or(auto_decimals + 2);
         let auto_decimals = auto_decimals.clamp(min_decimals, max_decimals);
+
+        if is_kb_editing {
+            let mut input = ui.input_mut();
+            let change = input.count_and_consume_key(Modifiers::NONE, Key::ArrowUp) as f64
+                - input.count_and_consume_key(Modifiers::NONE, Key::ArrowDown) as f64;
```

Comment:
The previous code also listened for `ArrowRight` and `ArrowLeft`, which matches the direction of most sliders, and is also a valid drag-axis.

Reply:
I thought I replied to this one a few days ago, but apparently I didn't actually post it.

The problem is that the left and right arrows can be used to move the caret when in keyboard editing mode. So we can't consume those keystrokes in `DragValue` itself.

To look at an existing accessible stepper/spinner implementation, I created a little web page with an `<input type="number">` element. In Chromium's implementation of that input type, at least on Windows, the up and down arrows adjust the number, but the left and right arrows move the caret.

Before this PR, `DragValue` was supposed to be able to listen for keypresses while in button mode, but that never actually happened, because as soon as the widget receives keyboard focus, it switches to keyboard editing mode. In an earlier iteration of my AccessKit branch, I modified `DragValue` to stay in button mode when it received the keyboard focus, unless the user pressed Enter, at which point it would switch to editing mode. This allows all arrows to be used to adjust the number. The problem with this is that the JAWS screen reader for Windows assumes that a spinner (the Windows name for this type of widget) is always in edit mode when it has the focus. (NVDA and Narrator seem to have no problem with a non-editable spinner, though.) And that previous change was also more drastic than what I'm doing now. So I figured that, all things considered, it's best to keep the current behavior of immediately going into edit mode on focus, and not respond to the left and right arrows. I'm open to other thoughts though.